test/e2e/clusterctl_upgrade.go

```diff
@@ -387,14 +390,48 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 			input.PreWaitForCluster(managementClusterProxy, testNamespace.Name, workLoadClusterName)
 		}
 
+		machineCRD := &apiextensionsv1.CustomResourceDefinition{}
+		if err := managementClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "machines.cluster.x-k8s.io"}, machineCRD); err != nil {
+			Expect(err).ToNot(HaveOccurred(), "failed to retrieve a machine CRD")
+		}
+
+		// Build GroupVersionKind for Machine resources
+		machineListGVK := schema.GroupVersionKind{
+			Group: machineCRD.Spec.Group,
+			Kind:  machineCRD.Spec.Names.ListKind,
+		}
+
+		// Sort the machineCRD.Spec.Versions slice
+		sort.Slice(machineCRD.Spec.Versions, func(i, j int) bool {
+			return machineCRD.Spec.Versions[i].Name < machineCRD.Spec.Versions[j].Name
+		})
+
```

Comment:
This should not bring any value. In my test-case it is already sorted like this:

```
❯ kubectl get crd machines.cluster.x-k8s.io -o yaml | yq '.spec.versions[].name'
v1alpha3
v1alpha4
v1beta1
```


```suggestion
```

Reply:
I would also drop it and just use the storage version to be honest. Not sure if just sorting alphabetically is the right thing to do for apiVersions btw :)

Reply:
Removed the sorting and used storage version